src/EdgeAnalysis.jl

```diff
@@ -237,13 +237,25 @@ function getkeyselectioninds(coreedgedomains,coreedgeinters,domains,inters)
             @inbounds push!(corerxnrangearray,index:index+length(inters[i].reactions))
             @inbounds push!(edgerxnrangearray,index+length(inters[i].reactions):index+length(coreedgeinters[i].reactions))
             @inbounds index += length(coreedgeinters[i].phase.reactions)
-
+            for (j,rxn) in enumerate(coreedgeinters[i].reactions)
+                @inbounds coreind = findfirst(isequal(rxn),inters[i].phase.reactions)
+                if coreind === nothing
+                    push!(edgerxninds,j+rxnindexedge)
+                else
+                    coretoedgerxnmap[coreind+rxnindexcore] = j+rxnindexedge
+                    push!(corerxninds,j+rxnindexedge)
+                end
+            end
+            @inbounds rxnindexcore += length(inters[i].reactions)
+            @inbounds rxnindexedge += length(coreedgeinters[i].reactions)
             @inbounds indend = length(inters[i].reactions)
-            @inbounds reactantindices[:,ind:ind+indend] = inters[i].rxnarray[1:4,:]
-            @inbounds productindices[:,ind:ind+indend] = inters[i].rxnarray[5:8,:]
+            @inbounds reactantindices[:,ind:ind+indend] = coreedgeinters[i].rxnarray[1:3,:]
+            @inbounds productindices[:,ind:ind+indend] = coreedgeinters[i].rxnarray[4:6,:]
```

Comment:
switch to four species add @views

Reply:
fixed

Reply:
This one as well, could you squash the fix?